### DISCUSSION TITLE
Can't use cacheDuration

### DISCUSSION THREAD

**Discussion by MoRadwan74** [TerriaJS/terriajs#5387] [Q&A] [answered]
Hello, I have a JSON file that contains some layers that I'm trying to show in Terria without caching the file in the browser. So, I added `"cacheDuration": "0s"` to each layer like that:
```
{
"name": "great_britain_points",
"type": "wms",
"cacheDuration": "0s",
"url": "http://localhost/geoserver/ows",
"layers": "cartoview_datastore:geonode:great_britain_points",
"opacity": 0.9
},
```

But the layers still caching, any thoughts??

**Comment by tephenavies** [comment, marked as answer]
HI @MoRadwan74. The property "cacheDuration" only applies to items that are proxied through terriajs-server `proxy/` endpoint. I think something else must be adding cache headers. Check the network requests that terriajs makes (using network tab of browser debugger - you can filter to "localhost/geoserver/ows" to make it easier to find the relevant requests) and take a look at the cache headers.

See https://developer.mozilla.org/en-US/docs/Web/HTTP/Caching & https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control to better understand the effect of specific caching headers.

If the caching is not in headers it could be geoserver or maybe a reverse proxy in front of geoserver that's adding caching.

**Reply by tephenavies** [reply]
If it is being proxied through terriajs-server the URL will show up in the network tab as "/proxy/http://localhost/geoserver/ows" or "/proxy/0s/http://localhost/geoserver/ows"

**Reply by MoRadwan74** [reply]
Yes, I forgot to talk about proxy. The JSON that I'm requesting is proxied and the request in the network tab looks like that `http://localhost:3001/proxy/http://localhost:7000/sda-catalog-layers.json` but it didn't contain the `cacheDuration` value, I even tried to make it something other than `0s` like `2m` but the request is still not changed.

Here are some screens for the request and headers.
![2](https://user-images.githubusercontent.com/30727109/113250146-3fff3c80-92c0-11eb-973e-494ac2134fcc.png)
![2](https://user-images.githubusercontent.com/30727109/113250043-10e8cb00-92c0-11eb-9ce6-d17b44e46631.png)

**Reply by MoRadwan74** [reply]
In the first image, you can find it still comes from the cache.
